policy_definitions/tagging/policy_assignment_qby_deploy_require_tags.tmpl.json

```diff
@@ -0,0 +1,20 @@
+{
+    "name": "QBY-Deploy-Require-Tags",
+    "type": "Microsoft.Authorization/policyAssignments",
+    "apiVersion": "2022-06-01",
+    "dependsOn": [],
+    "properties": {
+      "description": "Automatically generated policy set for Enforces existence of a tag in resource groups and all resources.",
+      "displayName": "QBY-Deploy-Require-Tags",
+      "enforcementMode": null,
+      "nonComplianceMessages": [],
```

Comment:
You should add nonComplianceMessages to make it easier for the user to know what is wrong.
See [documentation](https://learn.microsoft.com/en-us/azure/governance/policy/concepts/assignment-structure#non-compliance-messages) on implementation. Best would be to include custom noncompliance message for the specific policy, so the user knows, which tag is missing.

Reply:
Changed as suggested.